public/pages/Configuration/Configuration.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useState } from 'react';
+import {
+  EuiFlexItem,
+  EuiPanel,
+  EuiTitle,
+  EuiFlexGrid,
+  EuiText,
+  EuiFieldNumber,
+  EuiSelect,
+  EuiFlexGroup,
+  EuiFormRow,
+  EuiForm,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiBottomBar,
+} from '@elastic/eui';
+import { useHistory } from 'react-router-dom';
+
+const Configuration = () => {
+  const timeUnits = [
+    { value: 'MINUTES', text: 'Minute(s)' },
+    { value: 'HOURS', text: 'Hour(s)' },
+  ];
+
+  const minutesOptions = [
+    { value: 'ONE', text: '1' },
+    { value: 'FIVE', text: '5' },
+    { value: 'TEN', text: '10' },
+    { value: 'THIRTY', text: '30' },
+  ];
+
+  const defaultTopN = '10';
+  const defaultWindowSize = minutesOptions[3].value;
+  const defaultTimeUnit = timeUnits[0].value;
+
+  const history = useHistory();
+
+  const [topNSize, setTopNSize] = useState(defaultTopN);
+  const [windowSize, setWindowSize] = useState(defaultWindowSize);
+  const [time, setTime] = useState(defaultTimeUnit);
+
+  const onTopNSizeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setTopNSize(e.target.value);
+  };
+
+  const onWindowSizeChange = (
+    e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLSelectElement>
+  ) => {
+    setWindowSize(e.target.value);
+  };
+
+  const onTimeChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setTime(e.target.value);
+  };
+
+  const MinutesBox = () => (
+    <EuiSelect
+      id="minutes"
+      required={true}
+      options={minutesOptions}
+      value={windowSize}
+      onChange={(e) => onWindowSizeChange(e)}
+    />
+  );
+
+  const HoursBox = () => (
+    <EuiFieldNumber
+      min={1}
+      max={24}
+      required={true}
+      value={windowSize}
+      onChange={(e) => onWindowSizeChange(e)}
+    />
+  );
+
+  const WindowChoice = time === timeUnits[0].value ? MinutesBox : HoursBox;
+
+  let changed;
+  if (topNSize !== defaultTopN) {
+    changed = 'topN';
+  } else if (windowSize !== defaultWindowSize) {
+    changed = 'windowSize';
+  }
+
+  let valid = false;
```

Comment:
we can declare valid as const to avoid accidentally modifying it later.
```
const isValid = (() => {
    const nVal = parseInt(topNSize, 10);
    if (nVal < 1 || nVal > 100) return false;
    if (time === timeUnits[0].value) return true;
    const windowVal = parseInt(windowSize, 10);
    return windowVal >= 1 && windowVal <= 24;
  })();
```